modules/client-handler/src/main/java/org/apache/ignite/client/handler/JdbcQueryEventHandlerImpl.java

```diff
@@ -228,16 +240,17 @@ public CompletableFuture<JdbcBatchExecuteResult> batchPrepStatementAsync(long co
         });
     }
 
-    private CompletableFuture<Long> executeAndCollectUpdateCount(long connectionId, String sql, Object[] arg) {
-        var context = createQueryContext(JdbcStatementType.UPDATE_STATEMENT_TYPE);
-
+    private CompletableFuture<Long> executeAndCollectUpdateCount(long connectionId, boolean autoCommit, String sql, Object[] arg) {
         JdbcConnectionContext connectionContext;
         try {
             connectionContext = resources.get(connectionId).get(JdbcConnectionContext.class);
         } catch (IgniteInternalCheckedException exception) {
             return CompletableFuture.failedFuture(new IgniteInternalException(Client.CONNECTION_ERR));
         }
 
+        Transaction tx = autoCommit ? null : connectionContext.getOrStartTransaction();
+        var context = createQueryContext(JdbcStatementType.UPDATE_STATEMENT_TYPE, tx);
```

Comment:
keyword `var` is not allowed in this context

Reply:
Done, thanks